core/pkg/eval/string_comparison_evaluation.go

```diff
@@ -0,0 +1,110 @@
+package eval
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// startsWithEvaluation checks if the given property starts with a certain prefix.
+// It returns 'true', if the value of the given property starts with the prefix, 'false' if not.
+// As an example, it can be used in the following way inside an 'if' evaluation:
+//
+//	{
+//	  "if": [
+//			{
+//				"starts_with": [{"var": "email"}, "user@faas"]
+//			},
+//			"red", null
+//			]
+//	}
+//
+// This rule can be applied to the following data object, where the evaluation will resolve to 'true':
+//
+// { "email": "user@faas.com" }
+//
+// Note that the 'starts_with'  evaluation rule must contain exactly two items, which both resolve to a
+// string value
+func (je *JSONEvaluator) startsWithEvaluation(values, _ interface{}) interface{} {
```

Comment:
I think we can get rid of the pointer receiver here. Yes, `fractionalEvaluation` also uses the same. But, the only usage is to access the logger and IMO this is not necessary - we should be able to inject a dedicated logger and make this independent (low coupling)

And going further, all custom operators can be aggregated similar to what we have here :) 

I think this is a good refactoring and if preferred can be a follow-up (including fractionalEvaluation )

Reply:
@bacherfl Yes, this is what I was proposing. And we can do the same for `fractionalEvaluation` where we can inject the logger and refer implementation where needed (similar to`StringComparisonEvaluator` usage )

Yeah, agree with the point on the follow-up. This will improve the readability & testability.

Reply:
https://github.com/open-feature/flagd/issues/667